### ISSUE TITLE
Why do you multiply an importance weight on the cross-entropy function of IWCDAN and IWDAN?

### ISSUE THREAD

**Issue by johnnylu305** [microsoft/Domain-Adaptation-with-Conditional-Distribution-Matching-and-Generalized-Label-Shift#1]
Hi, nice to meet you. The equation 8 in the paper does not contain an importance weight w, but I see an importance weight w in your cross-entropy loss function for the classifier. Why do you multiply an importance weight w on the cross-entropy function of IWCDAN and IWDAN? Do I miss or misunderstand something?

**Comment by johnnylu305** [comment]
@rems75 Hi, could you give me a hint about it?

**Comment by rxian** [comment]
> Why do you multiply an importance weight w on the cross-entropy function of IWCDAN and IWDAN?

I guess this is a way to instruct the classifier to do better on classes that contain more examples on the target domain.

**Comment by rems75** [comment]
Hi @johnnylu305. Sorry for the slow reply, I somehow never got notified. 
That was an experiment to see whether upweighting the "larger" classes in the target domain helped the classifier as suggested by rxian. In the end, it did not change anything so we dropped it from the paper. 
It should have very little impact on the results of the algo.

**Comment by johnnylu305** [comment]
Hi, the explanation from rxian makes sense. Also, thank you to clarify the paper part.

**Comment by rxian** [comment]
> In the end, it did not change anything so we dropped it from the paper.

[This paper](http://proceedings.mlr.press/v97/byrd19a.html) studied and discussed why importance weighting may not help when applied to classification loss under label shift.
